### PR TITLE
Update log-destination to use the enumeration for LoggingDestination.

### PR DESCRIPTION
**Pull request by rolfyone**
Addresses #1641

Signed-off-by: Paul Harris <paul.harris@consensys.net>

### CODE REVIEW

File: artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java [hunk 1]
```diff
@@ -54,9 +55,10 @@
   @CommandLine.Option(
       names = {LOG_DESTINATION_OPTION_NAME},
       paramLabel = "<LOG_DESTINATION>",
-      description = "Whether a logger is added for the console, the log file, or both",
+      description =
+          "Whether a logger is added for the console, the log file, or both (Valid values: ${COMPLETION-CANDIDATES})",
       arity = "1")
-  private String logDestination = DEFAULT_LOG_DESTINATION;
+  private LoggingDestination logDestination = LoggingDestination.get(DEFAULT_LOG_DESTINATION);
```
Comment (ajsutton) [inline artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java]: Wouldn't it make more sense to just use:
```suggestion
  private LoggingDestination logDestination = LoggingDestination.DEFAULT_BOTH;
```

File: artemis/src/test/resources/complete_config.yaml [hunk 2]
```diff
@@ -32,7 +32,7 @@ eth1-endpoint: "http://localhost:8545"
 # logging
 log-color-enabled: True
 log-include-events-enabled: True
-log-destination: "both"
+log-destination: "DEFAULT_BOTH"
```
Comment (ajsutton) [inline artemis/src/test/resources/complete_config.yaml]: This probably should specify something that's not the default so we can tell it actually got picked up.

File: artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java [hunk 3]
```diff
@@ -81,7 +83,7 @@ public boolean isLogIncludeEventsEnabled() {
   }
 
   public String getLogDestination() {
-    return logDestination;
+    return logDestination.name();
```
Comment (ajsutton) [inline artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java]: The code that uses this value expects the `key` not the enum `name()`.  Currently key is internal only so a getter would be needed. I think it happens to work out at the moment but we should actually refer to the same value.  I suspect the `key` value is no longer relevant and we should just use `.valueOf` to look up the matching value.  Or just pass `LoggingDestination` enum through instead of going to `String` and back if the class path will let us do that.

File: artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java (hunk 1 above)
Comment (rolfyone) [reply artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java]: just means its not set off the default - i can if you prefer...

File: artemis/src/test/resources/complete_config.yaml (hunk 2 above)
Comment (rolfyone) [reply artemis/src/test/resources/complete_config.yaml]: its tested in other tests, but we can...

Comment (rolfyone) [reply artemis/src/test/resources/complete_config.yaml]: this test is actually super fragile, i'm going to avoid changing that complete_config file...

File: artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java (hunk 3 above)
Comment (rolfyone) [reply artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java]: the project dependencies are pretty ugly. i'll have another go.

Comment (ajsutton) [reply artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java]: I wouldn't try adjusting dependencies to make it happen.  Just make sure we consistently use either name or key to go to string and back.

File: artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java (hunk 1 above)
Comment (ajsutton) [reply artemis/src/main/java/tech/pegasys/artemis/cli/options/LoggingOptions.java]: The `DEFAULT_LOG_DESTINATION` constant could also be of type `LoggingDestination`.  Although I really don't know why we've extracted all this stuff as constants and will likely do a pass through to inline them all when I get a chance.  It's so much easier to read in besu where the strings are used directly in the `@Option` annotations.

File: artemis/src/test/resources/complete_config.yaml (hunk 2 above)
Comment (rolfyone) [reply artemis/src/test/resources/complete_config.yaml]: refactored tests a little bit so that we can see all logging options are picked up from configuration file. I think I've addressed the core of this concern.

File: artemis/src/test/resources/complete_config.yaml [hunk 4]
```diff
@@ -32,7 +32,7 @@ eth1-endpoint: "http://localhost:8545"
 # logging
 log-color-enabled: True
 log-include-events-enabled: True
-log-destination: "both"
+log-destination: "BOTH"
```
Comment (ajsutton) [inline artemis/src/test/resources/complete_config.yaml]: nit: Can probably undo this change - it should be case insensitive but lowercase will definitely be more common (and more likely to break since the enum name is uppercase)

**Review by ajsutton — APPROVED** [review]
LGTM.

File: artemis/src/test/resources/complete_config.yaml (hunk 4 above)
Comment (rolfyone) [reply artemis/src/test/resources/complete_config.yaml]: except the tests don't like it...

File: artemis/src/test/java/tech/pegasys/artemis/cli/AbstractBeaconNodeCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.ArgumentCaptor;
+import tech.pegasys.artemis.util.config.ArtemisConfiguration;
+
+public class AbstractBeaconNodeCommandTest {
```
Comment (mbaxter) [inline artemis/src/test/java/tech/pegasys/artemis/cli/AbstractBeaconNodeCommandTest.java]: Should this actually be `abstract`? :D 
```suggestion
public abstract class AbstractBeaconNodeCommandTest {
```

File: artemis/src/test/java/tech/pegasys/artemis/cli/options/LoggingOptionsTest.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.artemis.cli.options.LoggingOptions.LOG_DESTINATION_OPTION_NAME;
+import static tech.pegasys.artemis.util.config.LoggingDestination.DEFAULT_BOTH;
+
+import org.junit.jupiter.api.Test;
+import tech.pegasys.artemis.cli.AbstractBeaconNodeCommandTest;
+import tech.pegasys.artemis.util.config.ArtemisConfiguration;
+import tech.pegasys.artemis.util.config.LoggingDestination;
+
+public class LoggingOptionsTest extends AbstractBeaconNodeCommandTest {
```
Comment (mbaxter) [inline artemis/src/test/java/tech/pegasys/artemis/cli/options/LoggingOptionsTest.java:25]: Nice call splitting this out 🎉

File: artemis/src/test/java/tech/pegasys/artemis/cli/options/LoggingOptionsTest.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    // This is important!
+    // If it defaults to "both" or some other value custom log4j configs get overwritten
+    beaconNodeCommand.parse(new String[0]);
+
+    final ArtemisConfiguration artemisConfiguration = getResultingArtemisConfiguration();
+    assertThat(artemisConfiguration.getLogDestination()).isEqualTo(DEFAULT_BOTH);
+  }
+
+  @Test
+  public void logDestination_ShouldAcceptFileAsDestination() {
```
Comment (mbaxter) [inline artemis/src/test/java/tech/pegasys/artemis/cli/options/LoggingOptionsTest.java]: (nit) I think we usually capitalize like:
```suggestion
  public void logDestination_shouldAcceptFileAsDestination() {
```

File: util/src/main/java/tech/pegasys/artemis/util/config/LoggingDestination.java [hunk 8]
```diff
@@ -11,7 +11,7 @@
  * specific language governing permissions and limitations under the License.
  */
 
-package tech.pegasys.teku.logging;
+package tech.pegasys.artemis.util.config;
```
Comment (mbaxter) [inline util/src/main/java/tech/pegasys/artemis/util/config/LoggingDestination.java:14]: Ultimately,`util` is the wrong place for logging-specific constants.  But fixing this issue will require a larger refactor.  Will discuss offline when / how to prioritize refactoring.

**Review by mbaxter — APPROVED** [review]
A few nits, but otherwise LGTM! 🚀

File: artemis/src/test/java/tech/pegasys/artemis/cli/AbstractBeaconNodeCommandTest.java (hunk 5 above)
Comment (rolfyone) [reply artemis/src/test/java/tech/pegasys/artemis/cli/AbstractBeaconNodeCommandTest.java]: yes :) it should!

File: artemis/src/test/java/tech/pegasys/artemis/cli/options/LoggingOptionsTest.java (hunk 7 above)
Comment (rolfyone) [reply artemis/src/test/java/tech/pegasys/artemis/cli/options/LoggingOptionsTest.java]: gah, i knew that and i still stuffed it up

File: util/src/main/java/tech/pegasys/artemis/util/config/LoggingDestination.java (hunk 8 above)
Comment (rolfyone) [reply util/src/main/java/tech/pegasys/artemis/util/config/LoggingDestination.java:14]: sounds good!
